domain/resource/state/resource.go

```diff
@@ -2034,11 +2040,15 @@ func (st *State) getResourceToSet(typeIDs typeIDs, charmID corecharm.ID, res res
 	if err != nil {
 		return setResource{}, "", errors.Capture(err)
 	}
+	revision := (*int)(nil)
+	if res.Revision > 0 {
```

Comment:
todo: `res.Revision >= 0` or  `res.Revision > -1`  zero  is a valid resource revision number.

Reply:
Changed this here as well, as it is the same logic https://github.com/juju/juju/blob/f0cd00db3ff1f4db14106a17d17fda71bb4b6b57/apiserver/facades/client/application/deployrepository.go#L232